Include package bootstrapping example

OdatNurd opened the pull request:
This includes sample code for a Dependency (or a regular package, if
you wanted to) that is capable of automatically taking part of itself
to create a whole new `sublime-package` or keep that file updated as
the dependency is updated.

In a dependency, this allows the dependency to contain Sublime
commands and resources that can be made available as a regular package
even if/when Package Control installs the dependency in the new `lib`
folder of the most recent Subliem Text 3 builds.

The example shows:

  * Masking resources so Sublime won't load them except in the
    bootstrapped package.
  * Checking for the existence of a package
  * Creating or updating a package based on version checks
  * Ensuring that the update did not stop half way through
  * Taking care of resources that temporarily go away doing an update
    (e.g. syntax definitions that might be in use).

The code in the folder isn't laid out the way that an actual
dependency would be laid out, since the structure depends on the
contents of the resource in question (and it would make for an extra
`all` directory that just seems more confusing than helpful).

OdatNurd committed:
Include package bootstrapping example

This includes sample code for a Dependency (or a regular package, if
you wanted to) that is capable of automatically taking part of itself
to create a whole new `sublime-package` or keep that file updated as
the dependency is updated.

In a dependency, this allows the dependency to contain Sublime
commands and resources that can be made available as a regular package
even if/when Package Control installs the dependency in the new `lib`
folder of the most recent Subliem Text 3 builds.

The example shows:

  * Masking resources so Sublime won't load them except in the
    bootstrapped package.
  * Checking for the existence of a package
  * Creating or updating a package based on version checks
  * Ensuring that the update did not stop half way through
  * Taking care of resources that temporarily go away doing an update
    (e.g. syntax definitions that might be in use).

The code in the folder isn't laid out the way that an actual
dependency would be laid out, since the structure depends on the
contents of the resource in question (and it would make for an extra
`all` directory that just seems more confusing than helpful).

OdatNurd commented:
This is essentially a vaguely anonymized version of code that I wrote explicitly for hyperhelp to be able to provide its commands and resources even if/when PC installs dependencies in the `lib` folder, where even syntaxes would not be loaded directly.

Along with making sure the code isn't overly stupid or making blatantly wrong assumptions about things, let me know where the verbiage I thew into the various README files seems unclear or needs more embellishment.

On the whole I think as a scrap this one is more advanced than the usual since it requires more knowledge about the internals of Sublime in general.

keith-hall commented on package_bootstrap/1_my_package/README.md:
```diff
@@ -0,0 +1,17 @@
+    dependency that is creating the bootstrap package; that is, the created
+    package inherits the version of the dependency that created it.
+
+ 2. Files are added with the same name that they are seen here, but any that
```
Probably this would read better as:
`2. Files are added with the same file name that they have here, but any that`...

keith-hall commented on package_bootstrap/core/README.md:
```diff
@@ -0,0 +1,21 @@
+This folder represents the contentds of your dependency package that aren't
```
small typo here - it should say `contents`

keith-hall commented on package_bootstrap/core/bootstrapper.py:
```diff
@@ -0,0 +1,232 @@
+# resources and commands, and the base name of the file within that package
+# that is responsible for loading the commands into Sublime when the package
+# is loaded.
+bootstrap_pkg = "1_my_package"
```
probably the readme should mention more details about renaming this, and why it starts with a number :)

OdatNurd replied:
The new commit below clarifies this in a new section in the top level README, explaining why it's named that way, how to change it, and that it's in your best interests to use something unique. There are also a bevy of other minor tweaks to word wrapping and extraneous spacing.  Along with the typos/changes you mentioned above, I appear to have pathologically spelled `bootstrapper` wrong about 5 times.

OdatNurd committed:
Documentation fixes and clarifications

This includes some missing information in the original commit that
better describes how things work in the sample.

In particular a description of what the bootstrapped package name
represents and how to change it was missing from the original version
of the code

Apart from this, various inconsistencies in word wrapping and spacing
have been resolved as well.

OdatNurd committed:
Fix an incorrect word

This fixes up a word that was correctly spelled but not the right word
for the job that wasn't noticed until the commit was pushed up,
because that's just how that has to work.
